src/test/java/com/transferwise/openbanking/client/api/vrp/RestVrpClientTest.java

```diff
@@ -836,6 +837,39 @@ private OBDomesticVRPDetails aDomesticVrpDetailsResponse() {
             .data(data);
     }
 
+    private static class GetDomesticVrpConsentResponseArgumentProvider implements ArgumentsProvider {
+
+        @Override
+        public Stream<? extends Arguments> provideArguments(ExtensionContext context) {
+            var mockDomesticVrpConsentAcceptResponse = getAcceptedResponse();
+            String jsonResponse = jsonConverter.writeValueAsString(
+                mockDomesticVrpConsentAcceptResponse);
+            return Stream.of(
+                Arguments.of(jsonResponse, mockDomesticVrpConsentAcceptResponse),
+                Arguments.of(
+                    jsonResponse.replace(
+                        "Authorised",
+                        "Revoked"),
+                    getRejectedResponse()));
+        }
+
+        private OBDomesticVRPConsentResponse getAcceptedResponse() {
+            OBDomesticVRPConsentResponseData acceptedData =
+                new OBDomesticVRPConsentResponseData()
+                    .consentId("vrp-consent-id")
+                    .status(StatusEnum.AUTHORISED);
+            return new OBDomesticVRPConsentResponse().data(acceptedData);
+        }
+
+        private OBDomesticVRPConsentResponse getRejectedResponse() {
+            OBDomesticVRPConsentResponseData rejectedData =
+                new OBDomesticVRPConsentResponseData()
+                    .consentId("vrp-consent-id")
+                    .status(StatusEnum.REJECTED);
+            return new OBDomesticVRPConsentResponse().data(rejectedData);
+        }
+    }
+
```

Comment:
Can't this be a method? Why you need a class?

Reply:
I moved it to class as I need those helper class like  getAcceptedResponse,  getRejectedResponse. 
Is there any advantage of using method over class here? 
Also, All tests in this file uses Argument Provider.

Reply:
It is fine, just curious why you choose class here. Method is easier to read than a class. These methods getAcceptedResponse and getRejectedResponse are used only once.